CMakeLists.txt

```diff
@@ -380,6 +392,7 @@ add_executable(tests
     tests/test_resolve_names.cpp
     tests/test_restrictions.cpp
     tests/test_service.cpp
+    tests/test_subscribe.cpp
     tests/test_tasks.cpp
     tests/test_timeout.cpp)
 
```

Comment:
Add test_subscribe.cpp only to `tests` executable when `HAS_VARIANT` is true. Then remove `#ifdef` in test code.